design_pim/overview.rst

```diff
@@ -0,0 +1,354 @@
+How to customize any frontend part of the application
+-----------------------------------------------------
+
+Introduction
+===================
+
+Akeneo PIM uses RequireJS and Backbone to manage modules and views in our application. On top of those two layers, we added our own layer: the form extension system. In this documentation, we will discover the basics of those three layers to be able to understand Akeneo PIM stack and customize it.
+
+Note:
+Since the 1.8 version, we migrated to webpack to manage our front dependencies. To ease the migration we decided to keep the historicals ``requirejs.yml`` files to give the community enough time to migrate their configuration.
+
+Create our first RequireJS module
+=================================
+
+RequireJS modules are the equivalent of php classes. They provide a way to declare dependencies to a module (like use statements) and expose a new module.
+
+Simple business logic
++++++++++++++++++++++
+
+As an example, we will create a module that will display in the console the current UI locale
+
+To begin here is the business logic of our component:
+
+.. code-block:: javascript
+    :linenos:
+
+    {
+        logLocale: function () {
+            console.log(UserContext.get('uiLocale'));
+        }
+    }
+
+
+Our newly created object has one method which log the current catalog locale using the UserContext module.
+Now that we have our business logic implemented, we need to tell the system that our module has a dependency to ``pim/user-context`` and that we are exposing this object.
+
+The real thing
+++++++++++++++
+
+Here is how to do it using RequireJS:
+
+.. code-block:: javascript
+    :linenos:
+
+    // src/Acme/Bundle/CustomBundle/Resources/public/js/locale_logger.js
+    define(
+        ['pim/user-context'],
+        function (UserContext) {
+            return {
+                logLocale: function () {
+                    console.log(UserContext.get('uiLocale'));
+                }
+            }
+        }
+    )
+
+
+The ``define`` method takes two arguments: the first one is an array of the dependencies we want to use (``pim/user-context`` in this case) and a callback that will be called every time our module will be requested. This callback will receive as argument the list dependencies we asked for. Note that you can name the arguments of the callback as you want.
```

Comment:
Please add where the pim/user-context is defined (the .yml and the .js files for real)

Reply:
As this is just an example and not relevant, I'm not sure we should precise it. This module can be registered anywhere